modules/xfem/src/actions/XFEMAction.C

```diff
@@ -62,7 +62,8 @@ XFEMAction::act()
   MooseSharedPointer<XFEMInterface> xfem_interface = _problem->getXFEM();
   if (xfem_interface == NULL)
   {
-    MooseSharedPointer<XFEM> new_xfem (new XFEM(_problem->getMooseApp(), _problem));
+    _pars.set<FEProblem *>("_fe_problem") = &*_problem;
```

Comment:
I think this is fine for now. We'll be able to clean some of this up with C++11. There's a "weak pointer" concept to handle these "ring reference" situations. If this fixes the issue, I'll be happy.